src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,2766 @@
+import Data.Serialize qualified as S
+import Data.Set qualified as Set
+import Data.Vector qualified as Vec
+import Data.Word
+import Lens.Micro.Platform
+
+import Proto.V2.Concordium.Types qualified as Proto
+import Proto.V2.Concordium.Types_Fields qualified as ProtoFields
+
+import Concordium.Crypto.EncryptedTransfers
+import Concordium.ID.Types
+
+import Concordium.Types
+import Concordium.Types.Accounts
+import Concordium.Types.Queries qualified as QueryTypes
+import Concordium.Types.Transactions qualified as Transactions
+
+import Concordium.Common.Time
+import Concordium.Common.Version
+import Concordium.Crypto.SignatureScheme (Signature (..), VerifyKey (..))
+import Concordium.ID.AnonymityRevoker qualified as ArInfo
+import Concordium.ID.IdentityProvider qualified as IpInfo
+import Concordium.Types.Accounts.Releases
+import Concordium.Types.Block (AbsoluteBlockHeight (..))
+import Concordium.Types.Execution
+import Concordium.Types.InvokeContract qualified as InvokeContract
+import Concordium.Types.Parameters qualified as Parameters
+import Concordium.Types.Updates qualified as Updates
+import Concordium.Wasm qualified as Wasm
+import Data.Text (Text, pack)
+import Data.Time (UTCTime)
+
+import Concordium.Client.GRPC
+import Concordium.Client.Runner.Helper
+import Control.Exception
+import Network.GRPC.Client
+import Network.GRPC.Client.Helpers hiding (Address)
+import Network.GRPC.HTTP2.ProtoLens
+import Network.HTTP2.Client
+import Web.Cookie qualified as Cookie
+
+import Concordium.GRPC2 hiding (Output')
+import Concordium.ID.AnonymityRevoker (createArInfo)
+import Concordium.ID.IdentityProvider (createIpInfo)
+import Concordium.ID.Parameters (createGlobalContext)
+import Concordium.Types.Accounts qualified as Concordium.Types
+import Concordium.Types.InvokeContract (ContractContext (ContractContext))
+import Concordium.Types.Parameters (CryptographicParameters)
+import Control.Concurrent.Async
+import Control.Monad.Reader
+import Control.Monad.State.Strict
+import Data.ByteString (ByteString)
+import Data.IORef
+import Data.Maybe (fromJust, maybeToList)
+import Data.ProtoLens (defMessage)
+import Data.ProtoLens.Field qualified as Field
+import Data.ProtoLens.Service.Types
+import Data.Sequence (Seq (Empty))
+import Data.String
+import Data.Text qualified as Text
+import Proto.V2.Concordium.Service qualified as CS
+import Proto.V2.Concordium.Types qualified as ProtoFields
+import Proto.V2.Concordium.Types_Fields qualified as Proto
+import Concordium.Crypto.SHA256 (Hash)
+
+-- |A helper function that serves as an inverse to `mkSerialize`,
+--
+-- Converts a protocol buffer wrapper type to a native Haskell type
+-- which is a member of `Serialize`.
+--
+-- More concretely, the wrapper type should be of the form
+--
+-- > message Wrapper {
+-- >    bytes value = 1
+-- > }
+--
+-- where the name @Wrapper@ can be arbitrary, but the @value@ field must exist,
+-- and it must have type @bytes@.
+deMkSerialize ::
```

Comment:
I believe it would be a bit misleading to prefix it with `mk`. Unless I am misunderstood, the `mk` prefix is of `Proto.make`-fame and refers to constructing a message with the `value` field, and `Serialize` refers to `S.encode`, hence `mkSerialize`.

The `de`-prefix is merely to indicate it as an inverse. If you still  think `mkDeserialize` is better, I will change it to that 🙂

Reply:
It's fine, although what you describe is not really the etymology of the name. `mk` is not to do with `Proto.make`.

I don't want to discuss naming further.